### PR TITLE
Fix `--custom-typeshed-dir` crash caused by #13155

### LINKED ISSUE #13295: Crash on master branch when using `--custom-typeshed-dir` with a relative path
**Crash Report**

If you try to check typeshed with stubtest using the mypy master branch, it crashes. This is a result of the changes made in #13155.

**Traceback**

```
(venv) C:\Users\alexw\coding\typeshed>python -m mypy.stubtest --check-typeshed --custom-typeshed-dir .
Traceback (most recent call last):
  File "C:\Users\alexw\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\alexw\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "c:\users\alexw\coding\mypy\mypy\stubtest.py", line 1650, in <module>
    sys.exit(main())
  File "c:\users\alexw\coding\mypy\mypy\stubtest.py", line 1646, in main
    return test_stubs(parse_options(sys.argv[1:]))
  File "c:\users\alexw\coding\mypy\mypy\stubtest.py", line 1508, in test_stubs
    modules = build_stubs(modules, options, find_submodules=not args.check_typeshed)
  File "c:\users\alexw\coding\mypy\mypy\stubtest.py", line 1381, in build_stubs
    res = mypy.build.build(sources=sources, options=options)
  File "c:\users\alexw\coding\mypy\mypy\build.py", line 186, in build
    result = _build(
  File "c:\users\alexw\coding\mypy\mypy\build.py", line 250, in _build
    manager = BuildManager(
  File "c:\users\alexw\coding\mypy\mypy\build.py", line 677, in __init__
    if os.path.commonpath((path, custom_typeshed_dir)) == custom_typeshed_dir:
  File "C:\Users\alexw\AppData\Local\Programs\Python\Python310\lib\ntpath.py", line 772, in commonpath
    raise ValueError("Can't mix absolute and relative paths") from None
ValueError: Can't mix absolute and relative paths
```

**To Reproduce**

- `cd` into a local clone of mypy, checkout the `master` branch
- Activate a virtual environment with an editable install of mypy.
- `cd` into a local clone of typeshed.
- run `python -m mypy.stubtest --check-typeshed --custom-typeshed-dir .`

**Your Environment**

- Mypy version used: commit cbabbf7f1450eb1098d947386bcac92e90ff5909
- Python version used: 3.10
- Operating system and version: Windows

Cc. @hauntsaninja, @podshumok, @ilevkivskyi

### PR DESCRIPTION
**Pull request by AlexWaygood**
### Description

This seems to fix #13295.

## Test Plan

I manually verified that this fixes #13295. ~~Not sure how or where to add a test.~~  I also added a test case to `test-data/unit/cmdline.test` which reproduces the crash on `master`, but passes with this patch applied.

### CODE REVIEW

File: test-data/unit/cmdline.test [hunk 1]
```diff
@@ -1465,3 +1465,16 @@ sys.path = path
...
+[file dir/stdlib/abc.pyi]
+[file dir/stdlib/builtins.pyi]
+[file dir/stdlib/sys.pyi]
+[file dir/stdlib/types.pyi]
+[file dir/stdlib/typing.pyi]
+[file dir/stdlib/typing_extensions.pyi]
+[file dir/stdlib/VERSIONS]
+[out]
+Failed to find builtin module mypy_extensions, perhaps typeshed is broken?
+== Return code: 2
```
Comment (AlexWaygood) [inline test-data/unit/cmdline.test:1480]:
```suggestion
== Return code: 2

```

File: mypy/build.py [hunk 2]
```diff
@@ -674,6 +674,7 @@ def __init__(
             if options.custom_typeshed_dir is not None:
                 # Check if module lives under custom_typeshed_dir subtree
                 custom_typeshed_dir = os.path.abspath(options.custom_typeshed_dir)
+                path = os.path.abspath(path)
                 if os.path.commonpath((path, custom_typeshed_dir)) == custom_typeshed_dir:
```
Comment (podshumok) [inline mypy/build.py:678]: I believe it still can through `ValueError` on Windows if path and `custom_typeshed_dir` live on different drives.

Comment (AlexWaygood) [reply mypy/build.py:678]: Good point (though I'm tempted to say that anybody trying to do that deserves what they're getting)

**Review by hauntsaninja — APPROVED** [review]
Thanks!

**Pull request merged**